reconcile/utils/terrascript_aws_client.py

```diff
@@ -2255,7 +2277,54 @@ def populate_tf_resource_sqs(self, spec):
             tf_resources.append(tf_resource)
 
         self.add_resources(account, tf_resources)
+    
+    def populate_tf_resource_sns(self, spec):
+        account = spec.provisioner_name
+        identifier = spec.identifier
+        common_values = self.init_values(spec)
+        output_prefix = spec.output_prefix
+        policy = common_values.get("inline_policy")
+        region = common_values.get("region") or self.default_regions.get(account)
+        
+        values = {}
+        if 'fifo_topic' in common_values.keys():
+            fifo_topic = common_values.get("fifo_topic")
+        else:
+            fifo_topic = False
+        if fifo_topic:
```

Comment:
What if the user already added a `.fifo` suffix? Should we error here? Or should we require the user to suffix the name with `.fifo` to make it explicit?

Reply:
Since the identifier is of type "$ref": "/common-1.json#/definitions/longIdentifier", which doesn't allow . q-validator actually would throw an error, for example:
```
"{'provider': 'sns', 'identifier': 'test-sns-1.fifo', 'defaults': '/terraform/sns/sns-1.yml', 'subscriptions': [{'protocol': 'email', 'endpoint': 'yihuang@redhat.com'}]} is not valid under any of the given schemas\n\nFailed validating 'oneOf' in schema['properties']['externalResources']['items']['properties']['resources']['items']:\n    {'oneOf': [{'$ref': '/aws/terraform-resource-2.yml'}
```
I thought about making this a string instead and defer it to the integration to handle this so we support both arn or name as input. But I think since we are asking tenant to create SNS topic through a-i and their arn really isn't visible. So I decided go with this. Also more consistent with all the existing identifier.